client/src/components/screens/Patient/components/FilesTab/components/tableColumns.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import intl from 'react-intl-universal';
+import { useDispatch } from 'react-redux';
+import { DownOutlined } from '@ant-design/icons';
+import { getPatientFileURL } from 'actions/patient';
+import {
+    Button, Dropdown, Menu,
+} from 'antd';
+import { PatientFileFormat,PatientResponse, } from 'helpers/search/types';
+import { sortByOldest } from 'helpers/utils'
+
+const getDropdownOption = (
+    format: string,
+    url: {
+      file: string;
+      index: string;
+    },
+    metadata: PatientResponse,
+    
+  ) => {
+    const dispatch = useDispatch();
+    const onClickInitiDownload = (url:string) => dispatch(getPatientFileURL(url));
+    const options = [
+    (
+      <Menu.Item>
+        <Button
+          className="link--underline"
+          onClick={() => onClickInitiDownload(url.file)}
+          target="_blank"
+          type="link"
+        >
+          { intl.get('screen.patient.details.file.download.file') }
+        </Button>
+      </Menu.Item>
+    ),
+    (
+      <Menu.Item>
+        <Button
+          className="link--underline"
+          download={`${metadata.task.runDate}_${metadata.aliquot.resource[0].sample[0].resource.external_id}.json`}
+          href={`data:text/json;charset=utf-8,${encodeURIComponent(
+            JSON.stringify(metadata, null, 4)
+          )}`}
+          type="link"
+        >
+          Metadata
+        </Button>
+      </Menu.Item>
+    ),];
+  
+    if (format === PatientFileFormat.cram || format === PatientFileFormat.vcf) {
+      options.push(
+        (
+          <Menu.Item>
+            <Button
+              className="link--underline"
+              onClick={() => onClickInitiDownload(url.index)}
+              target="_blank"
+              type="link"
+            >
+  
+              Index
+            </Button>
+          </Menu.Item>
+        ),
+      );
+    }
+  
+  
+    return options
+  };
+
+export const filesColumns = [
+    {
+      intlKey: 'screen.patient.details.file.name',
+      name: 'title',
+    },
+    {
+      intlKey: 'screen.patient.details.file.type',
+      name: 'type',
+    },
+    {
+      intlKey: 'screen.patient.details.file.format',
+      name: 'format',
+    },
+    {
+      intlKey: 'screen.patient.details.file.size',
+      name: 'size',
+    },
+    {
+      intlKey: 'screen.patient.details.file.sample',
+      name: 'sample',
+    },
+    {
+      intlKey: 'screen.patient.details.file.aliquot',
+      name: 'aliquot',
+    },
+    {
+      intlKey: 'screen.patient.details.file.prescription',
+      name: 'prescription',
+      sorter: (columnInfoA: any, columnInfoB: any) => columnInfoA - columnInfoB,
```

Comment:
`columInfoA` is absolutely no more telling than `a`. More over, try to find another type instead of `any`.

Reply:
je pense que a et b sont ok pour un operation mathématique élémentaire comme ca. Dans la doc de Mozilla c'est même 'elem1' et 'elem2'